File: TaskMaster_API/TaskMaster/Model/Domain/ProjectData/IssueReport.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

using Microsoft.Identity.Client;

using TaskMaster.Model.API.ProjectData;
using TaskMaster.Model.API.UserData;
using TaskMaster.Model.Domain.UserData;

namespace TaskMaster.Model.Domain.ProjectData;

public class IssueReport
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int IssueReportId { get; set; }

    [Required]
    public string Title { get; set; }

    [Required]
    public int Priority { get; set; }

    [Required]
    public int Severity { get; set; }

    [Required]
    public int IssueType { get; set; }

    [Required]
    public string Details { get; set; }

    [Required]
    public bool Fixed { get; set; }

    public DateTime DateResolved { get; set; }
    
    public string Notes { get; set; }

    [Required]
    [ForeignKey("User")]
    public int UserId { get; set; }

    [Required]
    [ForeignKey("Project")]
    public int ProjectId { get; set; }

    public User User { get; set; }
    public Project Project { get; set; }    
}

File: App/TaskMaster/TaskMaster/Models/IssueReport.swift
//
//  IssueReport.swift
//  TaskMaster
//
//  Created by Joshua Root on 12/4/23.
//

import SwiftData
import Foundation

@Model
class IssueReport {
    var title: String
    var priority: Int
    var severity: Int
    var issueType: Int
    var details: String
    var fixed: Bool
    var dateResolved: Date?
    var notes: String
    var project: Project

    init(title: String, priority: Int, severity: Int, issueType: Int, details: String, fixed: Bool, dateResolved: Date?, notes: String, project: Project) {
        self.title = title
        self.priority = priority
        self.severity = severity
        self.issueType = issueType
        self.details = details
        self.fixed = fixed
        self.dateResolved = dateResolved
        self.notes = notes
        self.project = project
    }

    
}
